### PR TITLE
Upgrade RN Version - JS files

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] ~The `README.md` reflects new **features**~
 - [x] ~The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code~
 - [x] ~New unit tests and integration tests have beed added~
 - [x] ~Code coverage is more or equal to the previous coverage~
 - [x] A visual inspection of the `Files changed` tab was made prior to submitting the pull request ensuring the style guide was followed
 - [x] ~[CI pipeline](https://app.circleci.com/pipelines/github/twilio/twilio-voice-react-native-app) is green~
 - [x] ~Included screenshot if necessary~

## Description

This PR merges the JS file changes to the feature branch for updating this project's RN version.

See the RN Upgrade Helper for details: https://react-native-community.github.io/upgrade-helper/?from=0.70.9&to=0.72.17

### Breakdown

- Update JS files.

## Validation

- N/A

## Additional Notes

N/A

## Contributing to Twilio

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: app/jest.config.js [hunk 1]
```diff
@@ -1,13 +1,3 @@
...
-const config = {
-  moduleFileExtensions: ['js', 'ts', 'tsx'],
+module.exports = {
   preset: 'react-native',
-  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
-  testPathIgnorePatterns: ['/node_modules/', '/e2e/'],
-  transformIgnorePatterns: [
-    'node_modules/(?!@react-navigation|@react-native|react-native)',
-  ],
-  collectCoverage: true,
```
Comment (kpchoy) [inline app/jest.config.js:10]: Why are these configs no longer needed?

File: app/metro.config.js [hunk 2]
```diff
@@ -1,26 +1,8 @@
...
+ * @type {import('metro-config').MetroConfig}
  */
-
-const defaultSourceExts =
-  require('metro-config/src/defaults/defaults').sourceExts;
-
-module.exports = {
-  resolver: {
-    sourceExts:
-      process.env.MY_APP_MODE === 'mocked'
```
Comment (kpchoy) [inline app/metro.config.js:14]: I thought this line was needed for the e2e tests to work?

Comment (mhuynh5757) [reply app/metro.config.js:14]: Good catch! I'll add them back in.

File: app/jest.config.js (hunk 1 above)
Comment (mhuynh5757) [reply app/jest.config.js:10]: I should add them back in actually, the upgrade helper added this as a new file and I modified ours to match. Good catch.

File: server/.gitignore [hunk 3]
```diff
@@ -9,4 +10,3 @@ coverage/
 # logs
 combined.log
 error.log
-
```
Comment (kpchoy) [inline server/.gitignore:12]: Did you mean to remove this line?

**Review by kpchoy — APPROVED** [review]

**Pull request merged**